common/src/main/java/com/wynntils/features/user/NpcDialogAutoProgressFeature.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.google.common.collect.ImmutableList;
+import com.wynntils.core.chat.ChatModel;
+import com.wynntils.core.config.Config;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.core.features.properties.StartDisabled;
+import com.wynntils.core.managers.Model;
+import com.wynntils.mc.event.KeyInputEvent;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.wc.event.NpcDialogEvent;
+import java.util.Optional;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import net.minecraft.network.protocol.game.ServerboundPlayerCommandPacket;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import org.lwjgl.glfw.GLFW;
+
+@StartDisabled
+public class NpcDialogAutoProgressFeature extends UserFeature {
+
+    public static NpcDialogAutoProgressFeature INSTANCE;
+
+    private final ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
+
+    private final Runnable sendShiftRunnable = () -> {
+        McUtils.sendPacket(new ServerboundPlayerCommandPacket(
+                McUtils.player(), ServerboundPlayerCommandPacket.Action.PRESS_SHIFT_KEY));
+    };
+
+    private ScheduledFuture<?> lastScheduledFuture = null;
+
+    @Config
+    public static int dialogAutoProgressTime = 2000; // Milliseconds
```

Comment:
I think the time should scale with the length of the message. Possibly with a constant offset. So instead two variables:
minimumTimeBeforeProgress = 1500;
extraTimePerWord = 50; 
and count the number of words (or spaces) in the dialog.
(I have no idea if 50 is a good number, you'll have to check that and tweak it a bit, I guess)